Dockerfile

```diff
@@ -1,18 +1,17 @@
-FROM messense/rust-musl-cross:x86_64-musl AS builder
+FROM rust:1-alpine AS builder
```

Comment:
We could preserve [rust-musl-cross](https://github.com/rust-cross/rust-musl-cross) but avoid hardcoding  of `x86_64`

Reply:
[rust-musl-cross](https://github.com/rust-cross/rust-musl-cross) do not have a latest tag. The problem I had using rust-musl-cross is that I got the following error:

```
93.29 error: linking with `cc` failed: exit status: 1
93.29   |
93.29   = note: LC_ALL="C" PATH="/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/bin:/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-gnu/bin/self-contained:/root/.cargo/bin:/usr/local/musl/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" VSLANG="1033" "cc" "/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crt1.o" "/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crti.o" "/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crtbegin.o" "/tmp/rustcTsYzzo/symbols.o" "/target/aarch64-unknown-linux-musl/release/deps/dotenv_linter-eb4be4d0c34e32e8.dotenv_linter.3c3bdfe25051bfbf-cgu.0.rcgu.o" "-Wl,--as-needed" "-L" "/target/aarch64-unknown-linux-musl/release/deps" "-L" "/target/release/deps" "-L" "/target/aarch64-unknown-linux-musl/release/build/ring-95d7f71a6cd6c593/out" "-L" "/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib" "-Wl,-Bstatic" "/tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib" "-lunwind" "-lc" "/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/libcompiler_builtins-ab250a9fa88191e3.rlib" "-Wl,-Bdynamic" "-Wl,--eh-frame-hdr" "-Wl,-z,noexecstack" "-nostartfiles" "-L" "/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib" "-L" "/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained" "-o" "/target/aarch64-unknown-linux-musl/release/deps/dotenv_linter-eb4be4d0c34e32e8" "-Wl,--gc-sections" "-static" "-no-pie" "-Wl,-z,relro,-z,now" "-nodefaultlibs" "/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crtend.o" "/root/.rustup/toolchains/stable-aarch64-unknown-linux-gnu/lib/rustlib/aarch64-unknown-linux-musl/lib/self-contained/crtn.o"
93.29   = note: /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.29           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib(montgomery_inv.o): Relocations in generic ELF (EM: 62)
93.30           /usr/bin/ld: /tmp/rustcTsYzzo/libring-3e46ac32f9195160.rlib: error adding symbols: file in wrong format
93.30           collect2: error: ld returned 1 exit status
93.30           
93.30 
93.33 error: could not compile `dotenv-linter` (bin "dotenv-linter") due to 1 previous error
```

I can't quite fix the issue. This is the Dockerfile I used:

```Dockerfile
FROM messense/rust-musl-cross:x86_64-musl AS builder

ARG TARGETARCH

WORKDIR /

COPY Cargo.toml Cargo.lock ./
COPY dotenv-lookup ./dotenv-lookup
COPY dotenv-linter ./dotenv-linter

RUN if [ "${TARGETARCH}" = "amd64" ]; \
  then export TARGET="x86_64-unknown-linux-musl"; \
  else export TARGET="aarch64-unknown-linux-musl"; \
  fi \
  && rustup target add "${TARGET}" \
  && cargo build --release --target "${TARGET}" \
  && cargo install --target "${TARGET}" --path dotenv-linter

FROM scratch
COPY --from=builder /root/.cargo/bin/dotenv-linter /
ENTRYPOINT ["/dotenv-linter"]
```

Reply:
I think it should work in case of using `TARGETARCH` in `FROM`:

```Dockerfile

FROM messense/rust-musl-cross:${TARGETARCH}-musl AS builder

ARG TARGETARCH

```

But probably need to ensure platform passed in CI is corresponds to existed in rust-musk-cross

Reply:
Updated in https://github.com/dotenv-linter/dotenv-linter/commit/b2141f733773a72e24a8affd71dc75971ed9369b